Automattic/jetpack
# Payments Block: Payment button not showing custom amount field when configured

Fixes 776-gh-Automattic/dotcom-manage

#### Changes proposed in this Pull Request:
The payment block wasn't taking into account the `buyer_can_change_amount` when building the membership URL. This caused the payment form opened by the payment button to never show the custom amount field.

Implemented the fix at the Gutenberg block level, in order to fix the issue I needed to add a new property to the payment block, this property will handle if the membership form should show the custom amount field or not. 

Discarded fetching the membership data when building the URL while rendering each payment button.

#### Other information:


#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Go to a Simple/Atomic/Jurassic site.
* Make the code available for testing as required by the corresponding infrastructure.
* Create or add a new post.
* Add a payment button and select or create a subscription that has the `
Enable customers to pick their own amount` field checked and mark it as a donation.
* Update the post and open the post.
* Click on the payment button, and you should see the option to enter a custom amount.
* The other types of subscriptions should continue to work as expected.

##### Test evidence
https://user-images.githubusercontent.com/1989914/187436485-f78c78ce-3442-424f-b9c5-4e305698d1aa.mp4
